### PR TITLE
new:dev:SDK-386:Add Retries and Delay Options in python-sdk

### PR DESCRIPTION
**Pull request by shekharsaurabh**
As part of API throttling phase-2 dev, we are extending the capability to add retries and delay options for the user which can be consumed in case of RetryableExceptions.

### CODE REVIEW

File: qds_sdk/qubole.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Qubole:
     """
 
     MIN_POLL_INTERVAL = 1
+    RETRIES_CAP = 10
```
Comment (yogesh2021) [inline qds_sdk/qubole.py]: Try to come up with optimal value of retries and delay considering we want to finish all retries within 5 minutes.

File: qds_sdk/qubole.py [hunk 2]
```diff
@@ -21,6 +21,8 @@ class Qubole:
     """
 
     MIN_POLL_INTERVAL = 1
+    RETRIES_CAP = 6
```
Comment (shekharsaurabh) [inline qds_sdk/qubole.py]: Calculated the values as follow- 
With backoff factor as 2 and retry count of 6, we will be retrying 5 times, till the retry_count is >1, decrementing it on each retry. Keeping base_delay as 10sec,
Thus, all retries will be completed within
10 + 20 + 40 + 80 + 160 = 310sec (around 5 minutes)

File: qds_sdk/connection.py [hunk 3]
```diff
@@ -165,6 +191,9 @@ def _handle_error(response):
         elif code == 449:
             sys.stderr.write(response.text + "\n")
             raise RetryWithDelay(response, "Data requested is unavailable. Retrying ...")
+        elif code == 429:
```
[the remark this answers is not in the document]
Comment (shekharsaurabh) [reply qds_sdk/connection.py]: Addressed

**Comment by chattarajoy** [comment]
@shekharsaurabh , please merge the latest unreleased to your branch so that we can merge the changes.

**Pull request merged**